projects/i18n/switch/options.ts

```diff
@@ -0,0 +1,19 @@
+import {inject, InjectionToken} from '@angular/core';
+import {LOCAL_STORAGE} from '@ng-web-apis/common';
+import {TuiLanguageName} from '@taiga-ui/i18n/interfaces';
+
+export type TuiLangChunkLoader = (lang: TuiLanguageName) => Promise<unknown>;
+
+export const TUI_LANGUAGE_STORAGE_KEY = new InjectionToken<string>(
+    `Default key for search value in storage`,
+    {
+        factory: () => `tuiLang`,
+    },
+);
+
+export const TUI_LANGUAGE_DEFAULT_STORAGE = new InjectionToken<Storage>(
```

Comment:
I was also thinking about abstracting the `Storage` but couldn't think of a reason to. Do you have an example where this might come in handy? Maybe `IonicStorage` but it has different interface — it works with promises. This is probably a bit excessive.

Reply:
fixed

Reply:
fixed